services/acct/acctmgr.go

```diff
@@ -20,12 +21,13 @@ import (
 // account manager communicate with various backends for signing transactions.
 type AccountManager struct {
 	service.Service
-	chain    *blockchain.BlockChain
-	cfg      *config.Config
-	db       legacydb.DB
-	info     *AcctInfo
-	utxoops  []*UTXOOP
-	watchers map[string]*AcctBalanceWatcher
+	chain           *blockchain.BlockChain
+	cfg             *config.Config
+	db              legacydb.DB
+	info            *AcctInfo
+	utxoops         []*UTXOOP
+	watchers        map[string]*AcctBalanceWatcher
+	autoCollectUtxo chan types.AutoCollectUtxo
```

Comment:
if the channel is live inside the accountMgr, then its lifecycle should be same with the accountMgr. please rethink how to mange your channel (when to create it && when to destroy it)